pkg/apis/v1beta1/nodepool.go

```diff
@@ -78,6 +79,40 @@ type Disruption struct {
 	// +kubebuilder:validation:Schemaless
 	// +optional
 	ExpireAfter NillableDuration `json:"expireAfter"`
+	// Budgets is a list of Budgets.
+	// If there are multiple active budgets, Karpenter uses
+	// the most restrictive maxUnavailable. If left undefined,
+	// this will default to one budget with a maxUnavailable to 10%.
+	// +kubebuilder:validation:XValidation:message="'crontab' must be set with 'duration'",rule="!self.all(x, (has(x.crontab) && !has(x.duration)) || (!has(x.crontab) && has(x.duration)))"
+	// +kubebuilder:default:={{maxUnavailable: "10%"}}
+	// +kubebuilder:validation:MaxItems=50
+	// +optional
+	Budgets []Budget `json:"budgets,omitempty" hash:"ignore"`
+}
+
+// Budget defines when Karpenter will restrict the
+// number of Node Claims that can be terminating simultaneously.
+type Budget struct {
+	// MaxUnavailable dictates how many NodeClaims owned by this NodePool
+	// can be terminating at once. It must be set.
+	// This only considers NodeClaims with the karpenter.sh/disruption taint.
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:default:="10%"
+	MaxUnavailable intstr.IntOrString `json:"maxUnavailable" hash:"ignore"`
+	// Crontab specifies when a budget begins being active,
+	// using the upstream cronjob syntax. If omitted, the budget is always active.
+	// Currently timezones are not supported.
+	// This is required if Duration is set.
+	// +kubebuilder:validation:Pattern:=`^(@(annually|yearly|monthly|weekly|daily|midnight|hourly))|((.*)\s(.*)\s(.*)\s(.*)\s(.*))$`
+	// +optional
+	Crontab *string `json:"crontab,omitempty" hash:"ignore"`
```

Comment:
Cronjob uses `schedule` https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/

Reply:
Yep, I think schedule may be misinterpreted, since the actual schedule of this budget is determined by the combination of this crontab and the duration field. One could argue that it should be something like `starts` or `beginnings` but I think a field name of `crontab` indicates the syntax of the string more deliberately.

Reply:
Crontab seems a bit awkward to me, since a crontab is a file with a schedule + command: https://man7.org/linux/man-pages/man5/crontab.5.html